sdks/python/apache_beam/pipeline.py

```diff
@@ -396,28 +405,46 @@ def replace_all(self, replacements):
     for override in replacements:
       self._check_replacement(override)
 
-  def run(self, test_runner_api=True):
-    """Runs the pipeline. Returns whatever our runner returns after running."""
+  def run(self, test_runner_api=True, runner=None, options=None):
+    """Runs the pipeline. Returns whatever our runner returns after running.
 
+    If another runner instance and options are provided, that runner will
+    execute the pipeline with the given options. If either of them is not set,
+    the default runner will run the pipeline with the original options
+    assigned to the pipeline. The usage is similar to directly invoking
+    `runner.run_pipeline(pipeline, options)`.
+    """
+    runner_in_use = self.runner
+    options_in_use = self._options
+    if runner and options:
+      runner_in_use = runner
+      options_in_use = options
+    elif not runner and options:
+      raise ValueError('Parameter runner is not given when parameter options '
+                       'is given.')
+    elif not options and runner:
+      raise ValueError('Parameter options is not given when parameter runner '
+                       'is given.')
     # When possible, invoke a round trip through the runner API.
     if test_runner_api and self._verify_runner_api_compatible():
       return Pipeline.from_runner_api(
           self.to_runner_api(use_fake_coders=True),
-          self.runner,
-          self._options).run(False)
+          runner_in_use,
+          options_in_use,
+          interactive=self.interactive).run(False)
```

Comment:
Did you find that this was necessary? I don't think we should change the signature of the `from_runner_api` call. The pipeline protobuf should contain all the necessary information... Though I'd defer to @robertwb on this.

Reply:
No, it's not necessary. On a second thought, I can just move the logic to determine `interactive` ad hoc in `run()` and put the `interactive` field as a parameter in the `run()` method. Then I don't even need to change the Pipeline constructor.

Also, I've added this to the interactive_runner for below use case:
`InteractiveRunner(underlying_runner=DataflowRunner()).run_pipeline(Pipeline(DataflowRunner()))`